### PR TITLE
Add `leading`, `tracking` and `line_break` properties for label component.

### LINKED ISSUE #7602: Set and get additional font properties on labels
Hi!
We have translated our game into many languages, some of which require the use of custom fonts. Each font has its own unique leading, which is necessary for the correct display of text, as well as for the proper functioning of the resource.get_text_metrics(...) function. Unfortunately, it is currently not possible to set or get the leading from the script.

Also nice to have same get/set functional for line break
Now we use this code
```

  local size = go.get("sign#info_label", "size")
  local metrics = resource.get_text_metrics(go.get("sign#info_label", "font"), message, {
      width = size.x,
      line_break =true,
      leading = 1
  })
```

**Describe the solution you'd like (REQUIRED):**
```
local leading = go.get("sign#info_label", "leading")
go.set("sign#info_label", "leading", 1)

local has_line_break = go.get("sign#info_label", "line_break")
go.set("sign#info_label", "line_break", false)
```

```
  local size = go.get("sign#info_label", "size")
  local metrics = resource.get_text_metrics(go.get("sign#info_label", "font"), message, {
      width = size.x,
      line_break = go.get("sign#info_label", "line_break"),
      leading = go.get("sign#info_label", "leading")
  })
```
or like in gui nodes
```
  local size = go.get("sign#info_label", "size")
  local metrics = label.get_text_metrics("sign#info_label", message)
```

### PR DESCRIPTION
**Pull request by AGulev**
Now it is possible to get/set `leading`, `tracking` and `line_break`properties for a label component:
```
local leading = go.get("#label", "leading")
leading = leading * 1.2
go.set("#label", "leading", leading)
go.animate("#label", "leading", go.PLAYBACK_LOOP_PINGPONG, 2, go.EASING_INSINE, 2)

local tracking = go.get("#label", "tracking")
tracking = tracking * 1.2
go.set("#label", "tracking", tracking)
go.animate("#label", "tracking", go.PLAYBACK_LOOP_PINGPONG, 2, go.EASING_INSINE, 2)

local line_break = go.get("#label", "line_break")
go.set("#label", "line_break", false)
```

Fix https://github.com/defold/defold/issues/7602

## PR checklist

* [x] Code
	* [x] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [x] Add comments where needed
* [x] Documentation
	* [x] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [x] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: engine/engine/src/test/label/label.script [hunk 1]
```diff
@@ -47,6 +47,11 @@ local function test_vector3(name, expected)
     assert(expected.z == vz)
 end
 
+local function test_single_value(name, expected)
+	local val = go.get("#label", name)
+
+	assert(expected == val)
+end
```
Comment (JCash) [inline engine/engine/src/test/label/label.script:54]: looks like the tabs are not 4 spaces ?

File: engine/gamesys/src/gamesys/components/comp_label.cpp [hunk 2]
```diff
@@ -73,14 +73,18 @@ namespace dmGameSystem
...
         const char*                 m_Text;
 
         uint16_t                    m_ComponentIndex;
         uint16_t                    m_Enabled : 1;
         uint16_t                    m_AddedToUpdate : 1;
         uint16_t                    m_UserAllocatedText : 1;
         uint16_t                    m_ReHash : 1;
-        uint16_t                    m_Padding : 4;
+        uint16_t                    m_LineBreak : 1;
+        uint16_t                    m_Padding : 3;
```
Comment (JCash) [inline engine/gamesys/src/gamesys/components/comp_label.cpp:87]: Old bug: There are 16 bits in an uint16_t.
The old padding assumed it was an 8 bit variable.
Also, no need for a name:
```
uint16_t :11;
```

File: engine/gamesys/src/gamesys/components/comp_label.cpp [hunk 3]
```diff
@@ -591,6 +601,21 @@ namespace dmGameSystem
...
+            return dmGameObject::PROPERTY_RESULT_OK;
+        }
+        else if (get_property == LABEL_PROP_TRACKING)
+        {
+            out_value.m_Variant = dmGameObject::PropertyVar(component->m_Tracking);
+            return dmGameObject::PROPERTY_RESULT_OK;
+        }
+        else if (get_property == LABEL_PROP_LINE_BREAK)
+        {
+            out_value.m_Variant = dmGameObject::PropertyVar(component->m_LineBreak == 1);
```
Comment (JCash) [inline engine/gamesys/src/gamesys/components/comp_label.cpp:616]: Coding style: In general, we consider ` != 0` to be true.

**Review by JCash — DISMISSED** [review]
lgtm. Some minor code style issues.

File: engine/engine/src/test/label/label.script (hunk 1 above)
Comment (AGulev) [reply engine/engine/src/test/label/label.script]: true. This file cointains the mix of both tabs and spaces. So I changed everything to 4 spaces.

File: engine/gamesys/src/gamesys/components/comp_label.cpp (hunk 2 above)
Comment (AGulev) [reply engine/gamesys/src/gamesys/components/comp_label.cpp]: Nice catch! I missed that!

File: engine/gamesys/src/gamesys/components/comp_label.cpp (hunk 3 above)
Comment (AGulev) [reply engine/gamesys/src/gamesys/components/comp_label.cpp]: Totally makes sense!

**Pull request merged**